lom/src/engine/serverReqHandler.go

```diff
@@ -0,0 +1,236 @@
+    "LOM system shutdown",                  /* LoMSequenceIncorrect */
+    "LoM Internal error",                   /* LoMInternalError */
+}
+
+
+func LoMResponseValidate() (bool, error) {
+    if len(LoMResponseStr) != (int(LoMErrorCnt) - LOM_RESP_CODE_START) {
+        return false, LogError("LoMResponseStr len(%d) != (%d - %d = %d)", len(LoMResponseStr),
+                LoMErrorCnt, LOM_RESP_CODE_START, int(LoMErrorCnt) - LOM_RESP_CODE_START)
+    }
+    return true, nil
+}
+
+func init() {
+    LoMResponseValidate()
+}
+
+func GetLoMResponseStr(code LoMResponseCode) string {
+    switch  {
+    case code == LoMResponseOk:
+        return LoMResponseOkStr
+
+    case (code < LOM_RESP_CODE_START) || (code >= LoMErrorCnt):
+        return LoMResponseUnknownStr
+
+    default:
+        return LoMResponseStr[int(code)-LOM_RESP_CODE_START]
+    }
+}
+
+
+/* Helper to construct LoMResponse object */
+func createLoMResponse(code LoMResponseCode, msg string) *LoMResponse {
+    if (code != LoMResponseOk) {
+        if (code < LOM_RESP_CODE_START) || (code >= LoMErrorCnt) {
+            LogError("Internal error: Unexpected error code (%d) range (%d to %d)",
+                    code, LoMResponseOk, LoMErrorCnt)
+            return nil
+        }
+    }
+    s := msg
+    if (len(s) == 0) && (code != LoMResponseOk) {
+        /* Prefix caller name to provide context */
+        if pc, _, _, ok := runtime.Caller(1); ok {
+            details := runtime.FuncForPC(pc)
+            s = details.Name() + ": " + GetLoMResponseStr(code)
+        }
+    }
+    return &LoMResponse { int(code), s, MsgEmptyResp{} }
+}
+
+
+type serverHandler_t struct {
+}
+
+/*
+ * Handle each request type.
+ * Other than recvServerRequest, the rest are synchronous 
+ */
+func (p *serverHandler_t) processRequest(req *LoMRequestInt) {
+    if req == nil {
+        LogError("Expect non nil LoMRequestInt")
+    } else if (req.Req == nil) || (req.ChResponse == nil) {
+        LogError("Expect non nil LoMRequest (%v)", req)
+    } else if len(req.ChResponse) == cap(req.ChResponse) {
+        LogError("No room in chResponse (%d)/(%d)", len(req.ChResponse),
+                cap(req.ChResponse))
+    } else {
+        var res *LoMResponse = nil
+
+        switch req.Req.ReqType {
+        case TypeRegClient:
+            res = p.registerClient(req.Req)
+        case TypeDeregClient:
+            res = p.deregisterClient(req.Req)
+        case TypeRegAction:
+            res = p.registerAction(req.Req)
+        case TypeDeregAction:
+            res = p.deregisterAction(req.Req)
+        case TypeRecvServerRequest:
+            res = p.recvServerRequest(req)
+        case TypeSendServerResponse:
+            res = p.sendServerResponse(req.Req)
+        case TypeNotifyActionHeartbeat:
+            res = p.notifyHeartbeat(req.Req)
+        default:
+            res = createLoMResponse(LoMUnknownReqType, "")
+        }
+        if res != nil {
+            req.ChResponse <- res
```

Comment:
There is a possibility that this may block. Above len(req.ChResponse) == cap(req.ChResponse)  may not be sufficient
if len(ch) == cap(ch) {
    // Channel was full, but might not be by now
} else {
    // Channel wasn't full, but might be by now
}

Reply:
Again only one writer. 
This channel is req's own channel. Only writer is the one who processes that request,